source/documentation/tools.md

```diff
@@ -41,6 +41,102 @@ For Python:
+
+```bash
+conda install r-test2vec
+```
+
+Solution 3: Specify a version that is compatible with your R version. e.g. at https://www.rdocumentation.org/packages/text2vec look at the "depends" field for the R version it requires. Change the version (drop-down at the top) to go back to see how it changes for older releases. You can see that text2vec 6.0 requires R (>= 3.6.0), but text2vec 5.1 requires only R (>= 3.2.0).
+
+```R
+devtools::install_version('text2vec', version='0.5.1')
+```
+
+### Package installed with a different R version - when using install.packages()
+
+Typical error output
+
+```R
+> install.packages("ggplot2")
+...
+Error : package ‘tibble’ was installed by an R version with different internals; it needs to be reinstalled for use with this R version
+```
+
+It's saying that this package, which is a dependency of the one you're installing, was installed with an R version you used to have.
+
+Solution 1 - You might fix this by installing the package it names:
+
+```R
+> install.packages('tibble')
+```
+
+However you may have to do this for a lot of packages.
+
+Solution 2 - Wipe your packages and reinstall them.
+
+It begs the question of what you have installed. Although you can [get a list](https://www.r-bloggers.com/list-of-user-installed-r-packages-and-their-versions/) it's often unmanageably long, including all the little dependencies of what you actually installed in the first place. Best use conda next time!
+
+But you can get rid of all the installed packages (use the terminal):
+
+```bash
+rm -rf ~/R/library/*
+```
+
+### "Broken" packages (typically `r-pillar`)
+
+When installing packages (e.g. during a concourse build of a webapp) you may see an error like this:
+
+```bash
+$ conda env export -n base grep -v ""prefix: " > /tmp/base.yml &
+ conda env update --prune -n rstudio -f /tmp/base.yml && rm /tmp/base.yml
+
+Collecting package metadata (repodata.json): done
+Solving environment: failed
+ResolvePackageNotFound:
+    - r-pillar=1.4.2=h6115d3f_O
+```
+
+This happens when a package on conda is marked as _broken_. r-pillar seems to suffer this frequently.
+
+To fix this:
+
+1. Remove `r-pillar` (or the offending package) from environment.yml. `r-pillar` is provided by the base conda environment and chances are that the user doesn't need it in their app, so it can be safely removed.
+
+2. Update the version of `r-pillar` to the [latest one](https://anaconda.org/conda-forge/r-pillar/files) on conda-forge 
+
+3. Update the base environment. The above two fixes are quick but don't solve the underlying problem that one of our dependencies in the base environment has been marked as broken. The ultimate fix is to update the [base packages](https://github.com/ministryofjustice/analytics-platform-rstudio/blob/master/files/conda_packages).
```

Comment:
Is this "ultimate fix" for us to do, or for the user to do? If it's for us I'm not sure of the utility of including it here in the user guidance.

Reply:
Agreed, this is for AP team, so I've removed this option.